src/tests/quantity.rs

```diff
@@ -31,6 +31,19 @@ storage_types! {
         Test::assert_eq(&V::one(), &m1.get::<kilogram>());
     }
 
+    #[test]
+    fn from_str() {
```

Comment:
Add some negative tests:

 * No separator.
 * Invalid number that does not parse.
 * Invalid abbreviation (junk, abbreviation from different unit).
 * Valid abbreviation followed by extra: `1 m asdf`.